lib/coordinates.rb

```diff
@@ -0,0 +1,59 @@
+class Coordinates
+  def initialize(coords)
+    @coords = coords.split(" ")
+  end
+
+  def are_diagonal?
+    not are_vertical? and not are_horizontal?
+  end
+
+  def are_vertical?
+    expected_length = @coords.length
+    y_coords = @coords.map do |coord|
+      coord[1..-1].to_i
```

Comment:
Curious... let's consider revisiting this to account for edge cases, such as "AA3 AA4", etc. for coordinates.

Reply:
This won't matter at this point. If a player enters "AA1 AB23" each coordinate will be tested to see if it exists on the board. i.e. it won't exist as a key in the cells hash, so these coordinates will fail to pass a valid_coordinate? test